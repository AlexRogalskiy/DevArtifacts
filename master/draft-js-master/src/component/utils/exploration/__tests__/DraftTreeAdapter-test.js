/**
 * Copyright (c) 2013-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 * @emails oncall+ui_infra
 * @format
 */

'use strict';

jest.disableAutomock();

const DraftTreeAdapter = require('DraftTreeAdapter');

const assertFromRawTreeStateToRawState = rawState => {
  expect(
    DraftTreeAdapter.fromRawTreeStateToRawState(rawState),
  ).toMatchSnapshot();
};

const assertFromRawStateToRawTreeState = rawState => {
  expect(
    DraftTreeAdapter.fromRawStateToRawTreeState(rawState),
  ).toMatchSnapshot();
};

test('must be able to convert from tree raw state with only root blocks to raw state', () => {
  const rawState = {
    blocks: [
      {
        key: 'A',
        text: 'Alpha',
        children: [],
      },
      {
        key: 'B',
        text: 'Beta',
        children: [],
      },
    ],
    entityMap: {},
  };

  assertFromRawTreeStateToRawState(rawState);
});

test('must be able to convert from tree raw state with nested blocks to raw state', () => {
  const rawState = {
    blocks: [
      {
        key: 'A',
        type: 'blockquote',
        text: '',
        children: [
          {
            key: 'B',
            text: 'Beta',
            type: 'header-one',
            children: [],
          },
          {
            key: 'C',
            text: 'Charlie',
            type: 'header-two',
            children: [],
          },
        ],
      },
    ],
    entityMap: {},
  };

  assertFromRawTreeStateToRawState(rawState);
});

test('must be able to convert from tree raw state with nested list blocks to raw state preserving lists depth', () => {
  const rawState = {
    blocks: [
      {
        key: 'A',
        type: 'ordered-list-item',
        text: 'Alpha',
        children: [
          {
            key: 'B',
            text: 'Beta',
            type: 'ordered-list-item',
            children: [
              {
                key: 'C',
                text: 'Charlie',
                type: 'ordered-list-item',
                children: [],
              },
            ],
          },
          {
            key: 'D',
            text: 'Delta',
            type: 'ordered-list-item',
            children: [],
          },
        ],
      },
    ],
    entityMap: {},
  };

  assertFromRawTreeStateToRawState(rawState);
});

test('must be able to convert from tree raw state with nested list blocks to raw state preserving lists depth only if type matches', () => {
  const rawState = {
    blocks: [
      {
        key: 'A',
        type: 'ordered-list-item',
        text: 'Alpha',
        children: [
          {
            key: 'B',
            text: 'Beta',
            type: 'ordered-list-item',
            children: [
              {
                key: 'C',
                text: 'Charlie',
                type: 'unordered-list-item',
                children: [],
              },
            ],
          },
        ],
      },
    ],
    entityMap: {},
  };

  assertFromRawTreeStateToRawState(rawState);
});

test('must be able to convert from raw state to raw tree state', () => {
  const rawState = {
    blocks: [
      {
        key: 'A',
        text: 'Alpha',
      },
      {
        key: 'B',
        text: 'Beta',
      },
      {
        key: 'C',
        text: 'Charlie',
      },
    ],
    entityMap: {},
  };

  assertFromRawStateToRawTreeState(rawState);
});

test('must be able to convert from raw state to raw tree state with nested trees based on lists depth', () => {
  const rawState = {
    blocks: [
      {
        key: 'A',
        text: 'Alpha',
        type: 'ordered-list-item',
        depth: 0,
      },
      {
        key: 'B',
        text: 'Beta',
        type: 'ordered-list-item',
        depth: 1,
      },
      {
        key: 'C',
        text: 'Charlie',
        type: 'ordered-list-item',
        depth: 2,
      },
      {
        key: 'D',
        text: 'Delta',
        type: 'ordered-list-item',
        depth: 1,
      },
    ],
    entityMap: {},
  };

  assertFromRawStateToRawTreeState(rawState);
});

test('must be able to convert from raw state to raw tree state with nested trees based on lists depth and attach nested blocks to closest depth parent', () => {
  const rawState = {
    blocks: [
      {
        key: 'A',
        text: 'Alpha',
        type: 'ordered-list-item',
        depth: 0,
      },
      {
        key: 'B',
        text: 'Beta',
        type: 'ordered-list-item',
        depth: 1,
      },
      {
        key: 'C',
        text: 'Charlie',
        type: 'ordered-list-item',
        depth: 1,
      },
      {
        key: 'D',
        text: 'Delta',
        type: 'ordered-list-item',
        depth: 2,
      },
    ],
    entityMap: {},
  };

  assertFromRawStateToRawTreeState(rawState);
});
