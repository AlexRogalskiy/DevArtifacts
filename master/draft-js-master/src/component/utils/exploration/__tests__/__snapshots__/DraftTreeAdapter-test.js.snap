// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must be able to convert from raw state to raw tree state 1`] = `
Object {
  "blocks": Array [
    Object {
      "children": Array [],
      "key": "A",
      "text": "Alpha",
    },
    Object {
      "children": Array [],
      "key": "B",
      "text": "Beta",
    },
    Object {
      "children": Array [],
      "key": "C",
      "text": "Charlie",
    },
  ],
  "entityMap": Object {},
}
`;

exports[`must be able to convert from raw state to raw tree state with nested trees based on lists depth 1`] = `
Object {
  "blocks": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "depth": 2,
              "key": "C",
              "text": "Charlie",
              "type": "ordered-list-item",
            },
          ],
          "depth": 1,
          "key": "B",
          "text": "Beta",
          "type": "ordered-list-item",
        },
        Object {
          "children": Array [],
          "depth": 1,
          "key": "D",
          "text": "Delta",
          "type": "ordered-list-item",
        },
      ],
      "depth": 0,
      "key": "A",
      "text": "Alpha",
      "type": "ordered-list-item",
    },
  ],
  "entityMap": Object {},
}
`;

exports[`must be able to convert from raw state to raw tree state with nested trees based on lists depth and attach nested blocks to closest depth parent 1`] = `
Object {
  "blocks": Array [
    Object {
      "children": Array [
        Object {
          "children": Array [],
          "depth": 1,
          "key": "B",
          "text": "Beta",
          "type": "ordered-list-item",
        },
        Object {
          "children": Array [
            Object {
              "children": Array [],
              "depth": 2,
              "key": "D",
              "text": "Delta",
              "type": "ordered-list-item",
            },
          ],
          "depth": 1,
          "key": "C",
          "text": "Charlie",
          "type": "ordered-list-item",
        },
      ],
      "depth": 0,
      "key": "A",
      "text": "Alpha",
      "type": "ordered-list-item",
    },
  ],
  "entityMap": Object {},
}
`;

exports[`must be able to convert from tree raw state with nested blocks to raw state 1`] = `
Object {
  "blocks": Array [
    Object {
      "key": "A",
      "text": "",
      "type": "blockquote",
    },
    Object {
      "key": "B",
      "text": "Beta",
      "type": "header-one",
    },
    Object {
      "key": "C",
      "text": "Charlie",
      "type": "header-two",
    },
  ],
  "entityMap": Object {},
}
`;

exports[`must be able to convert from tree raw state with nested list blocks to raw state preserving lists depth 1`] = `
Object {
  "blocks": Array [
    Object {
      "depth": 0,
      "key": "A",
      "text": "Alpha",
      "type": "ordered-list-item",
    },
    Object {
      "depth": 1,
      "key": "B",
      "text": "Beta",
      "type": "ordered-list-item",
    },
    Object {
      "depth": 2,
      "key": "C",
      "text": "Charlie",
      "type": "ordered-list-item",
    },
    Object {
      "depth": 1,
      "key": "D",
      "text": "Delta",
      "type": "ordered-list-item",
    },
  ],
  "entityMap": Object {},
}
`;

exports[`must be able to convert from tree raw state with nested list blocks to raw state preserving lists depth only if type matches 1`] = `
Object {
  "blocks": Array [
    Object {
      "depth": 0,
      "key": "A",
      "text": "Alpha",
      "type": "ordered-list-item",
    },
    Object {
      "depth": 1,
      "key": "B",
      "text": "Beta",
      "type": "ordered-list-item",
    },
    Object {
      "depth": 0,
      "key": "C",
      "text": "Charlie",
      "type": "unordered-list-item",
    },
  ],
  "entityMap": Object {},
}
`;

exports[`must be able to convert from tree raw state with only root blocks to raw state 1`] = `
Object {
  "blocks": Array [
    Object {
      "key": "A",
      "text": "Alpha",
    },
    Object {
      "key": "B",
      "text": "Beta",
    },
  ],
  "entityMap": Object {},
}
`;
